packages/idyll-document/src/utils/index.js

```diff
@@ -238,12 +239,48 @@ export const filterIdyllProps = (props, filterInjected) => {
 export const getData = (arr, datasets = {}) => {
   const pluck = (acc, val) => {
     const nameValue = getProperties(val).name.value;
-    acc[nameValue] = datasets[nameValue];
+    const sourceValue = getProperties(val).source.value;
+    const async = getProperties(val).async
+      ? getProperties(val).async.value
+      : false;
+    if (async) {
+      const initialValue = getProperties(val).initialValue
+        ? JSON.parse(getProperties(val).initialValue.value)
+        : [];
+
+      let dataPromise = new Promise(res => res(initialValue));
+
+      if (typeof fetch !== 'undefined') {
+        dataPromise = fetch(sourceValue)
+          .then(res => {
+            if (sourceValue.endsWith('.csv')) {
+              return res
+                .text()
+                .then(resString => parse(resString))
```

Comment:
We should use the same options as the parsing that happens in the CLI so there aren't unexpected differences. There we call `parse(inputString, { cast: true, columns: true })`

Reply:
updated